src/hp/derp/server.rs

```diff
@@ -368,26 +305,192 @@ pub(crate) struct ServerInfo {
 use super::client_conn::ClientBuilder;
 // TODO: need a types module
 use super::client_conn::Packet;
+use super::client_conn::PeerConnState;
+
+use tokio::sync::mpsc;
+use tokio_util::sync::CancellationToken;
+
+pub(crate) struct ServerActor<C, R, W, P>
+where
```

Comment:
Not sure if it is worth it here, but you can DRY these bounds using a "module" trait.

```rust
trait ServerTypes {
  type C: Conn,
  type R: AsyncRead + Unpin + Send + Sync + 'static,
  type W: AsyncWrite + Unpin + Send + Sync + 'static,
  type P: PacketForwarder
}

pub(crate) struct ServerActor<ST: ServerTypes>
```

Not sure if it is worth it here, but if you get more types and more constraints it would definitely be worth it.